File: app/src/main/java/com/jaynesmaxtv/app/SplashActivity.java
package com.jaynesmaxtv.app;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.os.Build;
import android.view.WindowInsetsController;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

public class SplashActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            getWindow().setDecorFitsSystemWindows(false);
            WindowInsetsController c = getWindow().getInsetsController();
            if (c != null) {
                c.hide(android.view.WindowInsets.Type.statusBars() | android.view.WindowInsets.Type.navigationBars());
                c.setSystemBarsBehavior(WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
            }
        } else {
            getWindow().getDecorView().setSystemUiVisibility(
                View.SYSTEM_UI_FLAG_FULLSCREEN | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
        }
        setContentView(R.layout.activity_splash);

        View logo = findViewById(R.id.logo_container);
        View tagline = findViewById(R.id.tv_tagline);
        View loadingDots = findViewById(R.id.loading_dots);

        Animation fadeIn = AnimationUtils.loadAnimation(this, R.anim.fade_in);
        Animation slideUp = AnimationUtils.loadAnimation(this, R.anim.slide_up);

        logo.startAnimation(fadeIn);

        new Handler().postDelayed(() -> {
            tagline.setVisibility(View.VISIBLE);
            tagline.startAnimation(slideUp);
        }, 600);

        new Handler().postDelayed(() -> {
            loadingDots.setVisibility(View.VISIBLE);
            Animation pulse = AnimationUtils.loadAnimation(this, R.anim.pulse);
            loadingDots.startAnimation(pulse);
        }, 1200);

        new Handler().postDelayed(() -> {
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
            finish();
        }, 3500);
    }
}

File: app/src/main/java/com/jaynestv/max/ui/splash/SplashActivity.kt
package com.jaynestv.max.ui.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import com.jaynestv.max.R
import com.jaynestv.max.databinding.ActivitySplashBinding
import com.jaynestv.max.ui.home.HomeActivity
import com.jaynestv.max.ui.login.LoginActivity
import com.jaynestv.max.utils.SessionManager

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySplashBinding
    private lateinit var session: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
        window.statusBarColor = android.graphics.Color.parseColor("#080808")
        window.navigationBarColor = android.graphics.Color.parseColor("#080808")

        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)
        session = SessionManager(this)

        // Lottie animation — rings + play button kama Azam Max
        binding.lottieView.apply {
            setAnimation(R.raw.splash_animation)
            repeatCount = 0
            speed = 1.2f
            playAnimation()
        }

        // App name inaonekana baada ya animation
        binding.appName.postDelayed({
            binding.appName.visibility = View.VISIBLE
            binding.tagline.visibility = View.VISIBLE
            binding.appName.startAnimation(AnimationUtils.loadAnimation(
                this@SplashActivity, android.R.anim.fade_in))
        }, 1200)

        // Navigate baada ya ms 3000
        binding.lottieView.postDelayed({
            val dest = if (session.isLoggedIn()) HomeActivity::class.java
                       else                       LoginActivity::class.java
            startActivity(Intent(this, dest))
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
            finish()
        }, 3000)
    }
}
